### PR TITLE
New soem

### PR DESCRIPTION
**Pull request by jungerj**
# soem update
Lennart updated the soem package to the latest version.

### CODE REVIEW

**Comment by jungerj** [comment]
Lennart could build the package using colcon.
I will fix it for the catkin build system.

**Comment by firesurfer** [comment]
I merged in the submodule pr into this PR. Worked for me locally with colcon

**Comment by firesurfer** [comment]
@tomlankhorst or @jungerj 
Can someone of you try building it with catkin soon?

**Comment by firesurfer** [comment]
Works now also with catkin

**Comment by firesurfer** [comment]
Tried it on the testbench. Crashes for me

File: soem/CMakeLists.txt [hunk 1]
```diff
@@ -1,86 +1,126 @@
...
+
+
+catkin_package(
+   INCLUDE_DIRS 
+        ./soem
+        ./soem/soem
+        ./soem/osal
+        ./soem/osal/${OS}
+        ./soem/oshw/${OS}
+   LIBRARIES soem)
```
Comment (firesurfer) [inline soem/CMakeLists.txt:56]: perhaps export pthread as dependency?

**Comment by firesurfer** [comment]
@tomlankhorst @jungerj @JohannesPankert 
I would suggest merging this PR now. Jonas showed that it works on the testbench + on the mabi arm.

**Pull request merged**